BootloaderCorePkg/Stage2/Stage2Support.c

```diff
@@ -667,6 +677,34 @@ BuildExtraInfoHob (
     PlatformUpdateHobInfo (&gLoaderPlatformInfoGuid, LoaderPlatformInfo);
   }
 
+  if ((FixedPcdGetBool (PcdPreOsCheckerEnabled)) && (PldPtr[10] == EFI_FVH_SIGNATURE)) {
+    FvHeader = (EFI_FIRMWARE_VOLUME_HEADER *) PldPtr;
+
+    // Find pre-OS checker FFS file
+    Status = GetFfsFileByType (FvHeader, EFI_FV_FILETYPE_FREEFORM, 0, &PreOsCheckerFile);
```

Comment:
Do you think it is better to let OsLoader to handler the details on how to find PreOS image ?
We want to make Stage2 simple and generic.  So can we let Stage2 pass the FV base to FV payload ?